### ISSUE TITLE
0.3.4(yanked): regression config expansion no longer happens

### ISSUE THREAD

**Issue by ssbarnea** [ansible-community/molecule-docker#57]
Apparently last release no longer works with configuration items that have expansions as they endup being used directly:
```
  msg: 'Error pulling image molecule_local/${MOLECULE_IMAGE:-centos:7} - 400 Client Error for http+docker://ssh/v1.41/images/create?tag=7%7D&fromImage=molecule_local%2F%24%7BMOLECULE_IMAGE: Bad Request ("invalid reference format: repository name must be lowercase")'
```

The previous one 0.3.3 does not have the same bug. Until we fix it, please add `molecule-docker!=0.3.4` to your dependencies.

Initially reported by @arxcruz

**Comment by ssbarnea** [comment]
Bisecting I was able to identify that this regression was introduced by #51

**Comment by ssbarnea** [comment]
Apparently there are other issues with newer release. I seen error below with **ansible 2.9.22**, which was not expected:
```
msg: the connection plugin 'community.docker.docker' was not found
```
That is caused by https://github.com/ansible-community/molecule-docker/pull/56 as `'community.docker` is not installed by it

The big question is if we want to enforce use of `'community.docker` collection or if we want to fallback to the older docker module included with ansible 2.9

For that second problem the workaround is to run `ansible-galaxy collection install community.docker`

**Comment by ssbarnea** [comment]
I yanked 0.3.4 release and due to the huge number of changes needed the next release will be 0.4.0 and we will not make any fixes to the current one (being unable to test them correctly).
